src/util/color/rgbcolor.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <QColor>
+#include <QtGlobal>
+
+#include "util/assert.h"
+#include "util/optional.h"
+
+namespace mixxx {
+
+// A pure 24-bit 0xxRRGGBB color code without an alpha channel.
+// We are using a separate typedef, because QRgb implicitly
+// includes an alpha channel whereas this type does not!
+typedef quint32 RgbColorCode;
+
+// A thin wrapper around QColor to represent optional,
+// opaque RGB colors with 8-bit per channel without an
+// alpha channel. It ensures that the alpha channel is set
+// to opaque for valid RGB colors and that invalid colors
+// are always represented by the same, well defined QColor,
+// namely QColor().
+// Apart from assignment this type is immutable.
+class RgbColor {
+  public:
+    RgbColor()
+            : m_internalCode(kInvalidInternalCode) {
+        DEBUG_ASSERT(!isValid());
+    }
+    /*non-explicit*/ RgbColor(RgbColorCode code)
+            : m_internalCode(codeToInternalCode(code)) {
+        DEBUG_ASSERT(isValid());
+    }
+    /*non-explicit*/ RgbColor(std::optional<RgbColorCode> optionalCode)
+            : m_internalCode(optionalCodeToInternalCode(optionalCode)) {
+        DEBUG_ASSERT(isValid() == static_cast<bool>(optionalCode));
+    }
+    /*non-explicit*/ RgbColor(QColor anyColor)
+            : m_internalCode(anyColorToInternalCode(anyColor)) {
+        DEBUG_ASSERT(isValid() == anyColor.isValid());
+    }
+
+    // Implicit conversion into the corresponding QColor.
+    operator QColor() const {
+        return internalCodeToColor(m_internalCode);
+    }
+
+    // Checks if the color is valid or represents "no color",
+    // i.e. is missing or undefined. If the corresponding color
+    // code is stored in a database then the colum value is NULL.
+    bool isValid() const {
```

Comment:
Can we also add has_value() and and operator bool. In this case this new class will be become compatible to std::without the extra byte vor validity book keeping.